In [1]:
# 加载训练数据

import polars as pl

In [2]:
train_data = pl.read_csv('../data/train.csv')
train_data

shape: (3_960, 82)
┌──────────┬────────────┬────────────┬────────────┬───┬────────────┬────────────┬───────────┬──────┐
│ id       ┆ Basic_Demo ┆ Basic_Demo ┆ Basic_Demo ┆ … ┆ SDS-SDS_To ┆ PreInt_Edu ┆ PreInt_Ed ┆ sii  │
│ ---      ┆ s-Enroll_S ┆ s-Age      ┆ s-Sex      ┆   ┆ tal_T      ┆ Hx-Season  ┆ uHx-compu ┆ ---  │
│ str      ┆ eason      ┆ ---        ┆ ---        ┆   ┆ ---        ┆ ---        ┆ terintern ┆ i64  │
│          ┆ ---        ┆ i64        ┆ i64        ┆   ┆ i64        ┆ str        ┆ et_…      ┆      │
│          ┆ str        ┆            ┆            ┆   ┆            ┆            ┆ ---       ┆      │
│          ┆            ┆            ┆            ┆   ┆            ┆            ┆ i64       ┆      │
╞══════════╪════════════╪════════════╪════════════╪═══╪════════════╪════════════╪═══════════╪══════╡
│ 00008ff9 ┆ Fall       ┆ 5          ┆ 0          ┆ … ┆ null       ┆ Fall       ┆ 3         ┆ 2    │
│ 000fd460 ┆ Summer     ┆ 9          ┆ 0          ┆ … ┆ 64         ┆ Summer     ┆ 0         ┆ 0    │
│ 00105258 ┆ Summer     ┆ 10         ┆ 1          ┆ … ┆ 54         ┆ Summer     ┆ 2         ┆ 0    │
│ 00115b9f ┆ Winter     ┆ 9          ┆ 0          ┆ … ┆ 45         ┆ Winter     ┆ 0         ┆ 1    │
│ 0016bb22 ┆ Spring     ┆ 18         ┆ 1          ┆ … ┆ null       ┆ null       ┆ null      ┆ null │
│ …        ┆ …          ┆ …          ┆ …          ┆ … ┆ …          ┆ …          ┆ …         ┆ …    │
│ ff8a2de4 ┆ Fall       ┆ 13         ┆ 0          ┆ … ┆ 50         ┆ Fall       ┆ 1         ┆ 1    │
│ ffa9794a ┆ Winter     ┆ 10         ┆ 0          ┆ … ┆ null       ┆ Winter     ┆ 0         ┆ null │
│ ffcd4dbd ┆ Fall       ┆ 11         ┆ 0          ┆ … ┆ 77         ┆ Fall       ┆ 0         ┆ 1    │
│ ffed1dd5 ┆ Spring     ┆ 13         ┆ 0          ┆ … ┆ 47         ┆ Spring     ┆ 1         ┆ 0    │
│ ffef538e ┆ Spring     ┆ 11         ┆ 0          ┆ … ┆ null       ┆ Spring     ┆ 1         ┆ null │
└──────────┴────────────┴────────────┴────────────┴───┴────────────┴────────────┴───────────┴──────┘

观测数据可知，存在null值，所以需要对数据进行筛选或者填充。

In [ ]:
count = 0
for id in train_data['id']:
    try:
        file_load = pl.read_parquet(f'/kaggle/input/child-mind-institute-problematic-internet-use/series_train.parquet/id={id}/part-0.parquet')
    except:
        count+=1
count

观测得到，parquet文件并不是每个id都有的

查看null的比例，来判断是否应该填充

得到不同结果的数量，可知null的比例较大，不能舍去

In [37]:
(train_data
 .select(pl.col('PCIAT-PCIAT_Total'))
 .group_by(train_data.get_column('sii'))
 .agg(pl.col('PCIAT-PCIAT_Total').min().alias('PCIAT-PCIAT_Totalmin'),
      pl.col('PCIAT-PCIAT_Total').max().alias('PCIAT-PCIAT_Totalmax'),
      pl.col('PCIAT-PCIAT_Total').len().alias('count'))
      .sort('sii'))

shape: (5, 4)
┌──────┬──────────────────────┬──────────────────────┬───────┐
│ sii  ┆ PCIAT-PCIAT_Totalmin ┆ PCIAT-PCIAT_Totalmax ┆ count │
│ ---  ┆ ---                  ┆ ---                  ┆ ---   │
│ i64  ┆ i64                  ┆ i64                  ┆ u32   │
╞══════╪══════════════════════╪══════════════════════╪═══════╡
│ null ┆ null                 ┆ null                 ┆ 1224  │
│ 0    ┆ 0                    ┆ 30                   ┆ 1594  │
│ 1    ┆ 31                   ┆ 49                   ┆ 730   │
│ 2    ┆ 50                   ┆ 79                   ┆ 378   │
│ 3    ┆ 80                   ┆ 93                   ┆ 34    │
└──────┴──────────────────────┴──────────────────────┴───────┘

填充空白值
可选的填充方法
- KNN K近邻法
- 机器学习预测填充

In [71]:
import pandas as pd
train = pd.read_csv('../data/train.csv')

# 查看目标变量的缺失分布
print(train['sii'].isnull().sum())

# 分析其他特征与目标变量的相关性
correlations = train.corr()
print(correlations['sii'])


1224
Basic_Demos-Age                           0.365990
Basic_Demos-Sex                          -0.100148
CGAS-CGAS_Score                          -0.085261
Physical-BMI                              0.221125
Physical-Height                           0.373530
                                            ...   
PCIAT-PCIAT_Total                         0.899681
SDS-SDS_Total_Raw                         0.229692
SDS-SDS_Total_T                           0.232982
PreInt_EduHx-computerinternet_hoursday    0.336526
sii                                       1.000000
Name: sii, Length: 70, dtype: float64


C:\Users\Lenovo1\AppData\Local\Temp\ipykernel_13244\658698037.py:8: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  correlations = train.corr()


In [48]:
correlations

Basic_Demos-Age  Basic_Demos-Sex  \
Basic_Demos-Age                                1.000000         0.064004   
Basic_Demos-Sex                                0.064004         1.000000   
CGAS-CGAS_Score                               -0.009994         0.075470   
Physical-BMI                                   0.492112         0.025793   
Physical-Height                                0.880274        -0.014454   
...                                                 ...              ...   
PCIAT-PCIAT_Total                              0.409559        -0.093648   
SDS-SDS_Total_Raw                              0.037317        -0.020122   
SDS-SDS_Total_T                                0.037115        -0.018508   
PreInt_EduHx-computerinternet_hoursday         0.441822        -0.000579   
sii                                            0.365990        -0.100148   

                                        CGAS-CGAS_Score  Physical-BMI  \
Basic_Demos-Age                               -0.009994      0.492112   
Basic_Demos-Sex                                0.075470      0.025793   
CGAS-CGAS_Score                                1.000000     -0.094074   
Physical-BMI                                  -0.094074      1.000000   
Physical-Height                               -0.023185      0.487097   
...                                                 ...           ...   
PCIAT-PCIAT_Total                             -0.070542      0.240858   
SDS-SDS_Total_Raw                             -0.145369      0.087655   
SDS-SDS_Total_T                               -0.148457      0.086185   
PreInt_EduHx-computerinternet_hoursday        -0.021596      0.296709   
sii                                           -0.085261      0.221125   

                                        Physical-Height  Physical-Weight  \
Basic_Demos-Age                                0.880274         0.748149   
Basic_Demos-Sex                               -0.014454         0.003245   
CGAS-CGAS_Score                               -0.023185        -0.076399   
Physical-BMI                                   0.487097         0.865662   
Physical-Height                                1.000000         0.833844   
...                                                 ...              ...   
PCIAT-PCIAT_Total                              0.420765         0.353048   
SDS-SDS_Total_Raw                              0.036687         0.075098   
SDS-SDS_Total_T                                0.036478         0.073482   
PreInt_EduHx-computerinternet_hoursday         0.399367         0.375851   
sii                                            0.373530         0.325938   

                                        Physical-Waist_Circumference  \
Basic_Demos-Age                                             0.609601   
Basic_Demos-Sex                                            -0.020013   
CGAS-CGAS_Score                                            -0.076116   
Physical-BMI                                                0.892149   
Physical-Height                                             0.688033   
...                                                              ...   
PCIAT-PCIAT_Total                                           0.327013   
SDS-SDS_Total_Raw                                           0.089831   
SDS-SDS_Total_T                                             0.091152   
PreInt_EduHx-computerinternet_hoursday                      0.371418   
sii                                                         0.272885   

                                        Physical-Diastolic_BP  \
Basic_Demos-Age                                      0.107960   
Basic_Demos-Sex                                      0.002076   
CGAS-CGAS_Score                                     -0.005292   
Physical-BMI                                         0.141795   
Physical-Height                                      0.115724   
...                                                       ...   
PCIAT-PCIAT_Total

采用均值填充等均不合理，首先采用 KNN 填充的方式

In [49]:
from sklearn.impute import SimpleImputer, KNNImputer

imputer = KNNImputer(n_neighbors=5)

numeric_cols = train.select_dtypes(include=['float64', 'int64']).columns

# 填充数据

imputed_data = imputer.fit_transform(train[numeric_cols])

train_imputed = pd.DataFrame(imputed_data, columns=numeric_cols)

train_imputed['sii'] = train_imputed['sii'].round().astype(int)



In [50]:
train_imputed

Basic_Demos-Age  Basic_Demos-Sex  CGAS-CGAS_Score  Physical-BMI  \
0                 5.0              0.0             51.0     16.877316   
1                 9.0              0.0             70.0     14.035590   
2                10.0              1.0             71.0     16.648696   
3                 9.0              0.0             71.0     18.292347   
4                18.0              1.0             69.4     26.713639   
...               ...              ...              ...           ...   
3955             13.0              0.0             60.0     16.362460   
3956             10.0              0.0             58.6     18.764678   
3957             11.0              0.0             68.0     21.441500   
3958             13.0              0.0             70.0     12.235895   
3959             11.0              0.0             62.0     19.318614   

      Physical-Height  Physical-Weight  Physical-Waist_Circumference  \
0               46.00             50.8                          23.0   
1               48.00             46.0                          22.0   
2               56.50             75.6                          24.8   
3               56.00             81.6                          25.4   
4               64.14            125.0                          33.6   
...               ...              ...                           ...   
3955            59.50             82.4                          25.0   
3956            53.50             76.4                          27.0   
3957            60.00            109.8                          28.6   
3958            70.70             87.0                          27.6   
3959            61.10             92.9                          30.4   

      Physical-Diastolic_BP  Physical-HeartRate  Physical-Systolic_BP  ...  \
0                      61.2                86.4                 110.6  ...   
1                      75.0                70.0                 122.0  ...   
2                      65.0                94.0                 117.0  ...   
3                      60.0                97.0                 117.0  ...   
4                      70.6                74.2                 125.8  ...   
...                     ...                 ...                   ...  ...   
3955                   71.0                70.0                 104.0  ...   
3956                   60.0                78.0                 118.0  ...   
3957                   79.0                99.0                 116.0  ...   
3958                   59.0                61.0                 113.0  ...   
3959                   64.8                65.4                 108.4  ...   

      PCIAT-PCIAT_16  PCIAT-PCIAT_17  PCIAT-PCIAT_18  PCIAT-PCIAT_19  \
0                4.0             4.0             4.0             2.0   
1                0.0             0.0             0.0             0.0   
2                0.0             2.0             2.0             1.0   
3                3.0             4.0             3.0             4.0   
4                2.6             2.0             1.6             1.6   
...              ...             ...             ...             ...   
3955             0.0             2.0             1.0             1.0   
3956             0.8             2.0             0.8             0.8   
3957             0.0             1.0             1.0             0.0   
3958             1.0             1.0             1.0             1.0   
3959             0.4             1.4             1.4             1.8   

      PCIAT-PCIAT_20  PCIAT-PCIAT_Total  SDS-SDS_Total_Raw  SDS-SDS_Total_T  \
0                4.0               55.0               48.4             62.2   
1                0.0                0.0               46.0             64.0   
2                1.0               28.0               38.0             54.0   
3                1.0               44.0               31.0             45.0   
4                1.4               40.0               42.0         

In [51]:
for col in train.columns:
    if col not in numeric_cols:
        train_imputed[col] = train[col]


In [52]:
train_imputed

Basic_Demos-Age  Basic_Demos-Sex  CGAS-CGAS_Score  Physical-BMI  \
0                 5.0              0.0             51.0     16.877316   
1                 9.0              0.0             70.0     14.035590   
2                10.0              1.0             71.0     16.648696   
3                 9.0              0.0             71.0     18.292347   
4                18.0              1.0             69.4     26.713639   
...               ...              ...              ...           ...   
3955             13.0              0.0             60.0     16.362460   
3956             10.0              0.0             58.6     18.764678   
3957             11.0              0.0             68.0     21.441500   
3958             13.0              0.0             70.0     12.235895   
3959             11.0              0.0             62.0     19.318614   

      Physical-Height  Physical-Weight  Physical-Waist_Circumference  \
0               46.00             50.8                          23.0   
1               48.00             46.0                          22.0   
2               56.50             75.6                          24.8   
3               56.00             81.6                          25.4   
4               64.14            125.0                          33.6   
...               ...              ...                           ...   
3955            59.50             82.4                          25.0   
3956            53.50             76.4                          27.0   
3957            60.00            109.8                          28.6   
3958            70.70             87.0                          27.6   
3959            61.10             92.9                          30.4   

      Physical-Diastolic_BP  Physical-HeartRate  Physical-Systolic_BP  ...  \
0                      61.2                86.4                 110.6  ...   
1                      75.0                70.0                 122.0  ...   
2                      65.0                94.0                 117.0  ...   
3                      60.0                97.0                 117.0  ...   
4                      70.6                74.2                 125.8  ...   
...                     ...                 ...                   ...  ...   
3955                   71.0                70.0                 104.0  ...   
3956                   60.0                78.0                 118.0  ...   
3957                   79.0                99.0                 116.0  ...   
3958                   59.0                61.0                 113.0  ...   
3959                   64.8                65.4                 108.4  ...   

      CGAS-Season  Physical-Season  Fitness_Endurance-Season  FGC-Season  \
0          Winter             Fall                       NaN        Fall   
1             NaN             Fall                       NaN        Fall   
2            Fall             Fall                      Fall        Fall   
3            Fall           Summer                    Summer      Summer   
4          Summer              NaN                       NaN         NaN   
...           ...              ...                       ...         ...   
3955       Spring             Fall                       NaN        Fall   
3956          NaN           Spring                       NaN      Spring   
3957       Spring           Winter                       NaN      Winter   
3958       Spring           Winter                       NaN      Spring   
3959          NaN           Winter                       NaN      Winter   

      BIA-Season  PAQ_A-Season  PAQ_C-Season  PCIAT-Season  SDS-Season  \
0           Fall           NaN           NaN          Fall         NaN   
1         Winter           NaN          Fall          Fall        Fall   
2            NaN           NaN        Summer          Fall        Fall   
3         Summer           NaN        Winter        Summer      Summer   
4            NaN        Summer           NaN

处理数据之后对数据进行编码

In [53]:
train.columns

Index(['id', 'Basic_Demos-Enroll_Season', 'Basic_Demos-Age', 'Basic_Demos-Sex',
       'CGAS-Season', 'CGAS-CGAS_Score', 'Physical-Season', 'Physical-BMI',
       'Physical-Height', 'Physical-Weight', 'Physical-Waist_Circumference',
       'Physical-Diastolic_BP', 'Physical-HeartRate', 'Physical-Systolic_BP',
       'Fitness_Endurance-Season', 'Fitness_Endurance-Max_Stage',
       'Fitness_Endurance-Time_Mins', 'Fitness_Endurance-Time_Sec',
       'FGC-Season', 'FGC-FGC_CU', 'FGC-FGC_CU_Zone', 'FGC-FGC_GSND',
       'FGC-FGC_GSND_Zone', 'FGC-FGC_GSD', 'FGC-FGC_GSD_Zone', 'FGC-FGC_PU',
       'FGC-FGC_PU_Zone', 'FGC-FGC_SRL', 'FGC-FGC_SRL_Zone', 'FGC-FGC_SRR',
       'FGC-FGC_SRR_Zone', 'FGC-FGC_TL', 'FGC-FGC_TL_Zone', 'BIA-Season',
       'BIA-BIA_Activity_Level_num', 'BIA-BIA_BMC', 'BIA-BIA_BMI',
       'BIA-BIA_BMR', 'BIA-BIA_DEE', 'BIA-BIA_ECW', 'BIA-BIA_FFM',
       'BIA-BIA_FFMI', 'BIA-BIA_FMI', 'BIA-BIA_Fat', 'BIA-BIA_Frame_num',
       'BIA-BIA_ICW', 'BIA-BIA_LDM', 'BIA-BIA_LST'

season 数据 
两个：season of enrollment， season of participation， 貌似没用

In [55]:
data_description = pd.read_csv('../data/data_dictionary.csv')
data_description

Instrument  \
0                           Identifier   
1                         Demographics   
2                         Demographics   
3                         Demographics   
4   Children's Global Assessment Scale   
..                                 ...   
76             Sleep Disturbance Scale   
77             Sleep Disturbance Scale   
78             Sleep Disturbance Scale   
79                        Internet Use   
80                        Internet Use   

                                     Field                       Description  \
0                                       id                  Participant's ID   
1                Basic_Demos-Enroll_Season              Season of enrollment   
2                          Basic_Demos-Age                Age of participant   
3                          Basic_Demos-Sex                Sex of participant   
4                              CGAS-Season           Season of participation   
..                                     ...                               ...   
76                              SDS-Season           Season of participation   
77                       SDS-SDS_Total_Raw                   Total Raw Score   
78                         SDS-SDS_Total_T                     Total T-Score   
79                     PreInt_EduHx-Season           Season of participation   
80  PreInt_EduHx-computerinternet_hoursday  Hours of using computer/internet   

               Type                        Values  \
0               str                           NaN   
1               str  Spring, Summer, Fall, Winter   
2             float                           NaN   
3   categorical int                           0,1   
4               str  Spring, Summer, Fall, Winter   
..              ...                           ...   
76              str  Spring, Summer, Fall, Winter   
77              int                           NaN   
78              int                           NaN   
79              str  Spring, Summer, Fall, Winter   
80  categorical int                       0,1,2,3   

                                         Value Labels  
0                                                 NaN  
1                                                 NaN  
2                                                 NaN  
3                                    0=Male, 1=Female  
4                                                 NaN  
..                                                ...  
76                                                NaN  
77                                                NaN  
78                                                NaN  
79                                                NaN  
80  0=Less than 1h/day, 1=Around 1h/day, 2=Around ...  

[81 rows x 6 columns]

预估重要的指标


In [56]:
correlations['sii']

Basic_Demos-Age                           0.365990
Basic_Demos-Sex                          -0.100148
CGAS-CGAS_Score                          -0.085261
Physical-BMI                              0.221125
Physical-Height                           0.373530
                                            ...   
PCIAT-PCIAT_Total                         0.899681
SDS-SDS_Total_Raw                         0.229692
SDS-SDS_Total_T                           0.232982
PreInt_EduHx-computerinternet_hoursday    0.336526
sii                                       1.000000
Name: sii, Length: 70, dtype: float64

In [57]:
correlations['PCIAT-PCIAT_Total']

Basic_Demos-Age                           0.409559
Basic_Demos-Sex                          -0.093648
CGAS-CGAS_Score                          -0.070542
Physical-BMI                              0.240858
Physical-Height                           0.420765
                                            ...   
PCIAT-PCIAT_Total                         1.000000
SDS-SDS_Total_Raw                         0.234432
SDS-SDS_Total_T                           0.237718
PreInt_EduHx-computerinternet_hoursday    0.374124
sii                                       0.899681
Name: PCIAT-PCIAT_Total, Length: 70, dtype: float64

In [61]:

sii_corr = correlations['PCIAT-PCIAT_Total'].drop('PCIAT-PCIAT_Total')
filtered_corr = sii_corr[(sii_corr > 0.1) | (sii_corr < -0.1)]
filtered_corr

Basic_Demos-Age                           0.409559
Physical-BMI                              0.240858
Physical-Height                           0.420765
Physical-Weight                           0.353048
Physical-Waist_Circumference              0.327013
Physical-Systolic_BP                      0.147081
FGC-FGC_CU                                0.287494
FGC-FGC_GSND                              0.146813
FGC-FGC_GSD                               0.160472
FGC-FGC_PU                                0.196006
FGC-FGC_SRL_Zone                         -0.148850
FGC-FGC_SRR_Zone                         -0.109682
FGC-FGC_TL                                0.136696
BIA-BIA_BMI                               0.248060
BIA-BIA_FFMI                              0.109694
BIA-BIA_Frame_num                         0.193631
PCIAT-PCIAT_01                            0.727959
PCIAT-PCIAT_02                            0.788802
PCIAT-PCIAT_03                            0.823336
PCIAT-PCIAT_04                 

相关性较强的指标
Basic_Demos-Age                           0.409559
Physical-BMI                              0.240858
Physical-Height                           0.420765
Physical-Weight                           0.353048
Physical-Waist_Circumference              0.327013
Physical-Systolic_BP                      0.147081
FGC-FGC_CU                                0.287494
FGC-FGC_GSND                              0.146813
FGC-FGC_GSD                               0.160472
FGC-FGC_PU                                0.196006
FGC-FGC_SRL_Zone                         -0.148850
FGC-FGC_SRR_Zone                         -0.109682
FGC-FGC_TL                                0.136696
BIA-BIA_BMI                               0.248060
BIA-BIA_FFMI                              0.109694
BIA-BIA_Frame_num                         0.193631

In [63]:
season_cols = [col for col in train.columns if 'Season' in col]
filter_data = train.drop(season_cols, axis=1)

In [64]:
filter_data

id  Basic_Demos-Age  Basic_Demos-Sex  CGAS-CGAS_Score  \
0     00008ff9                5                0             51.0   
1     000fd460                9                0              NaN   
2     00105258               10                1             71.0   
3     00115b9f                9                0             71.0   
4     0016bb22               18                1              NaN   
...        ...              ...              ...              ...   
3955  ff8a2de4               13                0             60.0   
3956  ffa9794a               10                0              NaN   
3957  ffcd4dbd               11                0             68.0   
3958  ffed1dd5               13                0             70.0   
3959  ffef538e               11                0              NaN   

      Physical-BMI  Physical-Height  Physical-Weight  \
0        16.877316             46.0             50.8   
1        14.035590             48.0             46.0   
2        16.648696             56.5             75.6   
3        18.292347             56.0             81.6   
4              NaN              NaN              NaN   
...            ...              ...              ...   
3955     16.362460             59.5             82.4   
3956     18.764678             53.5             76.4   
3957     21.441500             60.0            109.8   
3958     12.235895             70.7             87.0   
3959           NaN              NaN              NaN   

      Physical-Waist_Circumference  Physical-Diastolic_BP  Physical-HeartRate  \
0                              NaN                    NaN                 NaN   
1                             22.0                   75.0                70.0   
2                              NaN                   65.0                94.0   
3                              NaN                   60.0                97.0   
4                              NaN                    NaN                 NaN   
...                            ...                    ...                 ...   
3955                           NaN                   71.0                70.0   
3956                          27.0                   60.0                78.0   
3957                           NaN                   79.0                99.0   
3958                           NaN                   59.0                61.0   
3959                           NaN                    NaN                 NaN   

      ...  PCIAT-PCIAT_16  PCIAT-PCIAT_17  PCIAT-PCIAT_18  PCIAT-PCIAT_19  \
0     ...             4.0             4.0             4.0             2.0   
1     ...             0.0             0.0             0.0             0.0   
2     ...             0.0             2.0             2.0             1.0   
3     ...             3.0             4.0             3.0             4.0   
4     ...             NaN             NaN             NaN             NaN   
...   ...             ...             ...             ...             ...   
3955  ...             0.0             2.0             1.0             1.0   
3956  ...             NaN             NaN             NaN             NaN   
3957  ...             0.0             1.0             1.0             0.0   
3958  ...             1.0             1.0             1.0             1.0   
3959  ...             NaN             NaN             NaN             NaN   

      PCIAT-PCIAT_20  PCIAT-PCIAT_Total  SDS-SDS_Total_Raw  SDS-SDS_Total_T  \
0                4.0               55.0                NaN              NaN   
1                0.0                0.0               46.0             64.0   
2                1.0               28.0               38.0             54.0   
3                1.0               44.0               31.0             45.0   
4                NaN                NaN                NaN              NaN   
...              ...                ...                ...              ...   
3955             0.0               32.0               35.0 

In [65]:
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline

In [66]:
preprocessor = ColumnTransformer(
    transformers=[
        ('num', StandardScaler(), numeric_cols)
    ]
)
X = preprocessor.fit_transform(filter_data)

In [68]:
X.shape

(3960, 70)

In [70]:
import numpy as np

import pandas as pd

import os

import re

from sklearn.base import clone

from sklearn.metrics import cohen_kappa_score

from sklearn.model_selection import StratifiedKFold

from scipy.optimize import minimize

from concurrent.futures import ThreadPoolExecutor

from tqdm import tqdm

import polars as pl

import polars.selectors as cs

import matplotlib.pyplot as plt

from matplotlib.ticker import MaxNLocator, FormatStrFormatter, PercentFormatter

import seaborn as sns



from sklearn.preprocessing import StandardScaler

import matplotlib.pyplot as plt

from keras.models import Model

from keras.layers import Input, Dense

from keras.optimizers import Adam

import torch

import torch.nn as nn

import torch.optim as optim



from colorama import Fore, Style

from IPython.display import clear_output

import warnings

from lightgbm import LGBMRegressor

from xgboost import XGBRegressor

from catboost import CatBoostRegressor

from sklearn.ensemble import VotingRegressor, RandomForestRegressor, GradientBoostingRegressor

from sklearn.impute import SimpleImputer, KNNImputer

from sklearn.pipeline import Pipeline

warnings.filterwarnings('ignore')

pd.options.display.max_columns = None

ModuleNotFoundError: No module named 'keras'

In [69]:
if np.any(np.isinf(train)):

    train = train.replace([np.inf, -np.inf], np.nan)



def quadratic_weighted_kappa(y_true, y_pred):

    return cohen_kappa_score(y_true, y_pred, weights='quadratic')



def threshold_Rounder(oof_non_rounded, thresholds):

    return np.where(oof_non_rounded < thresholds[0], 0,

                    np.where(oof_non_rounded < thresholds[1], 1,

                             np.where(oof_non_rounded < thresholds[2], 2, 3)))



def evaluate_predictions(thresholds, y_true, oof_non_rounded):

    rounded_p = threshold_Rounder(oof_non_rounded, thresholds)

    return -quadratic_weighted_kappa(y_true, rounded_p)

NameError: name 'np' is not defined